src/interp.cc

```diff
@@ -1214,6 +1214,25 @@ ValueTypeRep<T> Xchg(ValueTypeRep<T> lhs_rep, ValueTypeRep<T> rhs_rep) {
   return rhs_rep;
 }
 
+// i(8,16,32,64) f(32,64) X (2,4,8,16) splat ==> v128
+template <typename T, typename V>
+ValueTypeRep<T> SimdSplat(V LaneData) {
+  union SimdBits{
```

Comment:
I think this typecast through a union is undefined behavior. I'd prefer we use `Bitcast` here, which uses memcpy under the hood, which I believe is OK. So something like:
```
V lanes[sizeof(T) / sizeof(V)];
...
T data = Bitcast<T>(lanes);
```

For this to work properly though, `Bitcast` will have to be modified to take its argument by rvalue reference: `Dst Bitcast(Src&& value)`

Reply:
Ok, I will update the PR.

BTW: 
I still don't understand why you say "this typecast through a union is undefined behavior", Would you please explain it with more details if possible?

Thanks!

Reply:
Done.